### ISSUE TITLE
Enforce UTF-8 no BOM on files

### ISSUE THREAD

**Issue by krwq** [dotnet/iot#2059]
We should remove BOM from all files and add some enforcement so we don't change it back by accident.

**Comment by gfoidl** [comment]
> some enforcement so we don't change it back by accident

Is `.editorconfig` enough for this?

**Comment by krwq** [comment]
yes, we currently enforce reverse :-) we need to convert all files as part of this issue :-)

**Comment by pgrawehr** [comment]
Do we have a documentation why this should be an improvement? I'm still fearing that removing the BOM will result in different editors/OSes breaking non-ascii characters in our code files. And we have quite a few of those.
